### ISSUE TITLE
Come up with a plan and design for search widgets

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#3149]
- [ ] come up with a list of screens that need search widgets
- [ ] figure out what sort and filter options we need, and whether they can be common to all filters (e.g. "alive", "animals", etc.)
- [ ] figure out how vanilla list widgets can be filtered/sorted
- [ ] write design doc
- [ ] write project plan

**Comment by fricy** [comment]
As per [reddit discussion](https://www.reddit.com/r/dwarffortress/comments/127x7pr/comment/jemwb3z/?context=3):

Justice screen:
search widget
filter widget: resident, visitor, animal, dead. default hide animals, dead. alphabetical sort

Military -> search to assign position (useful for nicknames)
(low priority, DTherapist is a decent substitute) 

I'll try to find more.

**Comment by fricy** [comment]
linking to #3225 

Military: possibly filter by need: Martial training

Hauling routes list get cluttered if you try to use them. Filter option would help, but a new widget would be better.

**Comment by fricy** [comment]
Bins/Barrels:

Filter by type/material/quality

default view is collapsed to item type for mass forbid/dumping, can filter to select individual items

**Comment by myk002** [comment]
Suggestion from u/Dragoon209 on [reddit](https://www.reddit.com/r/dwarffortress/comments/12zob2m/comment/jhtiql5/?utm_source=share&utm_medium=web2x&context=3):

Along with the search bars, being able to maintain the position of a long list (like the justice screen in an old fort) would be awesome. Scrolling down through 500 creatures (including many dead) is very annoying.

Lastly, is it possible to enhance the existing searches for trade and the like to allow searching for meta tags? maybe something like 'wear:X' 'decorated:true' 'artifact:false' etc.

**Comment by myk002** [comment]
from u/BeerNTacos on [reddit](https://www.reddit.com/r/dwarffortress/comments/133fa83/comment/ji9ufx0/?utm_source=share&utm_medium=web2x&context=3):

For v50+, are there any mods or DFHack command that impact the artifacts menu on the map screen? If you're in a big and/or old world, it's absolutely tedious scrolling through it to find particular items. Of course something like a search function would be best, but I would take anything I could get to help improve that menu's functionality, especially since using that menu is the only way to obtain some artifacts, even if you go search the area it's in beforehand.

Maybe something like filters, so you can focus on one kind of item at a time, like rings, shirts, ridings, etc.?

Or perhaps we could have something where we click on a site on the map and the artifact window will display only the rumors associated to that site? Feeling that, perhaps a separate menu that could lists sites which could then modify the artifact screen to limit the choices to that site?

**Comment by a7x492** [comment]
Hi, this issue looks very ambitious but is there any chance a very simple search widget could be added to the "Work Orders" tab and that this could be prioritized since it should be simple and very impactful? There are already similar search widgets, for example the one at the top of the "Tasks" tab. Being able to search the Work Orders in the same way as we already can with Tasks would be a lifesaver for me.

**Comment by myk002** [comment]
The main work of this issue has already been completed. I think I was just keeping it open because of some of the "stretch goals" in the comments.

> very simple search widget could be added to the "Work Orders" tab 

I wish it were simple : ( The work orders tab cannot be filtered since there is no intermediate list for the UI. If we "filter" the displayed work orders, it is equivalent to deleting the work orders from the game and then (hopefully) reinstating them when the filter is removed. However, since the game has no guarantees about when it might *access* that list, the list is never safe to filter.

In short, the work orders screen is too brittle for search/filter functionality. It would need a full replacement UI. There is some discussion about related ideas here: https://github.com/DFHack/dfhack/issues/1910
